throughput/spam.go

```diff
@@ -128,48 +140,56 @@ func (s *Spammer) Spam(ctx context.Context, sh SpamHelper, terminate bool, symbo
 	}
 
 	// distribute funds
-	accounts, factories, err := s.distributeFunds(ctx, feePerTx, sh)
+	accounts, factories, err := s.distributeFunds(cctx, feePerTx, sh)
 	if err != nil {
 		return err
 	}
 
-	// create issuers
-	issuers, err := s.createIssuers(parser, ruleFactory)
-	if err != nil {
+	if err := s.run(cctx, cli, ruleFactory, sh, factories, feePerTx, terminate); err != nil {
 		return err
 	}
 
-	cctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	for _, issuer := range issuers {
-		issuer.Start(cctx)
-	}
-
-	// start logging
-	s.tracker.startPeriodicLog(ctx, cli)
+	utils.Outf("{{yellow}}waiting for issuers to return{{/}}\n")
+	s.issuerWg.Wait()
 
-	// broadcast transactions
-	err = s.broadcast(cctx, sh, factories, issuers, feePerTx, terminate)
-	cancel()
+	maxUnits, err = chain.EstimateUnits(ruleFactory.GetRules(time.Now().UnixMilli()), actions, s.authFactory)
 	if err != nil {
-		s.tracker.stop()
 		return err
 	}
+	return s.returnFunds(ctx, cli, maxUnits, sh, accounts, factories, symbol)
```

Comment:
```suggestion
		// Use the original context, so that we attempt to return funds before exiting after
		// receiving user interrupt
		return s.returnFunds(ctx, cli, maxUnits, sh, accounts, factories, symbol)
```